invokeai/frontend/install/invokeai_update.py

```diff
@@ -92,44 +102,30 @@ def get_extras():
 
 
 def main():
-    versions = get_versions()
-    released_versions = [x for x in versions if not (x["draft"] or x["prerelease"])]
-    prerelease_versions = [x for x in versions if not x["draft"] and x["prerelease"]]
-    latest_release = released_versions[0]["tag_name"] if len(released_versions) else None
-    latest_prerelease = prerelease_versions[0]["tag_name"] if len(prerelease_versions) else None
-
     if invokeai_is_running():
         print(":exclamation: [bold red]Please terminate all running instances of InvokeAI before updating.[/red bold]")
         input("Press any key to continue...")
         return
 
+    latest_release, latest_prerelease = get_pypi_versions()
+
     welcome(latest_release, latest_prerelease)
 
-    tag = None
-    branch = None
-    release = None
-    choice = Prompt.ask("Choice:", choices=["1", "2", "3", "4"], default="1")
+    release = latest_release
+    choice = Prompt.ask("Choice:", choices=["1", "2", "3"], default="1")
 
     if choice == "1":
         release = latest_release
     elif choice == "2":
         release = latest_prerelease
     elif choice == "3":
-        while not tag:
-            tag = Prompt.ask("Enter an InvokeAI tag name")
-    elif choice == "4":
-        while not branch:
-            branch = Prompt.ask("Enter an InvokeAI branch name")
+        release = Prompt.ask("Enter an InvokeAI version name")
```

Comment:
If the user presses return here without typing the version name the code will attempt to install an empty version, leading to:

```
Enter an InvokeAI version name: 
🤞 Upgrading to 


ERROR: Could not find a version that satisfies the requirement invokeai[xformers]== (from versions: 2.2.4.5, 2.2.4.6, 2.2.4.7, 2.2.5, 2.3.0a0, 2.3.0a1, 2.3.0a2, 2.3.0a3, 2.3.0rc3, 2.3.0rc4, 2.3.0rc5, 2.3.0rc6, 2.3.0rc7, 2.3.0, 2.3.1rc4, 2.3.1, 2.3.1.post1, 2.3.1.post2, 2.3.2, 2.3.2.post1, 2.3.3rc1, 2.3.3, 2.3.4a0, 2.3.4rc1, 2.3.4, 2.3.4.post1, 2.3.5rc1, 2.3.5, 2.3.5.post1, 2.3.5.post2, 3.0.0, 3.0.1rc1, 3.0.1rc2, 3.0.1, 3.0.1.post1, 3.0.1.post2, 3.0.1.post3, 3.0.2a1, 3.0.2rc1, 3.0.2, 3.0.2.post1, 3.1.0, 3.1.1rc1, 3.1.1, 3.2.0, 3.3.0, 3.3.0.post1, 3.3.0.post2, 3.3.0.post3, 3.4.0rc2, 3.4.0rc3, 3.4.0rc4, 3.4.0, 3.4.0.post1, 3.4.0.post2, 3.5.0rc1, 3.5.0rc2)
ERROR: No matching distribution found for invokeai[xformers]==
❗ Upgrade failed
```

I'd suggest creating a little while loop that keeps calling `Prompt.ask` until the user enters a valid release name contained in the `versions` dict that you create in line 57.

Reply:
Here's a bit of code that implements this logic:
```
diff --git a/invokeai/frontend/install/invokeai_update.py b/invokeai/frontend/install/invokeai_update.py
index cde9d8fd1..32d51907e 100644
--- a/invokeai/frontend/install/invokeai_update.py
+++ b/invokeai/frontend/install/invokeai_update.py
@@ -58,7 +58,7 @@ def get_pypi_versions():
     versions.sort(key=LooseVersion, reverse=True)
     latest_version = [v for v in versions if "rc" not in v][0]
     latest_release_candidate = [v for v in versions if "rc" in v][0]
-    return latest_version, latest_release_candidate
+    return latest_version, latest_release_candidate, versions
 
 
 def welcome(latest_release: str, latest_prerelease: str):
@@ -107,7 +107,7 @@ def main():
         input("Press any key to continue...")
         return
 
-    latest_release, latest_prerelease = get_pypi_versions()
+    latest_release, latest_prerelease, versions = get_pypi_versions()
 
     welcome(latest_release, latest_prerelease)
 
@@ -119,7 +119,12 @@ def main():
     elif choice == "2":
         release = latest_prerelease
     elif choice == "3":
-        release = Prompt.ask("Enter an InvokeAI version name")
+        while True:
+            release = Prompt.ask("Enter an InvokeAI version name")
+            release.strip()
+            if release in versions:
+                break
+            print(f":exclamation: [bold red]'{release}' is not a recognized InvokeAI release.[/red bold]")
 
     extras = get_extras()
```

Reply:
Good catch! Updated this logic